jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantDisplay/model.js

```diff
@@ -1,85 +1,198 @@
-    'LinearGenomeViewPlugin',
-  ).exports
+                        if (self.renderProps().config.color1.value != color){
+                           self.renderProps().config.color1.set(color)
 
-  return types
-    .compose(
-      'WidgetDisplay',
-      BaseLinearDisplay,
-      types.model({
-        type: types.literal('ExtendedVariantDisplay'),
-        configuration: ConfigurationReference(configSchema),
-      }),
-    )
+                           const { centerLineInfo } = getContainingView(self)
+                           const { refName, assemblyName, offset } = centerLineInfo
+                           const centerBp = Math.round(offset) + 1
 
-    .actions(self => ({
-      selectFeature(feature) {
-        var extendedVariantDisplayConfig
-        if(getContainingTrack(self).configuration.metadata.value.extendedVariantDisplayConfig){
-            extendedVariantDisplayConfig = getContainingTrack(self).configuration.metadata.value.extendedVariantDisplayConfig
-        }
-        else{
-            extendedVariantDisplayConfig = []
-        }
+                           const region = {
+                              start: centerBp,
+                              end: (centerBp || 0) + 1,
+                              refName,
+                              assemblyName,
+                           }
 
-        var message
-        if (getContainingTrack(self).configuration.metadata.value.message){
-            message = getContainingTrack(self).configuration.metadata.value.message
-        }
-        else{
-            message = ""
-        }
-        const trackId = getContainingTrack(self).configuration.trackId
-        const session = getSession(self)
-        var widgetId = 'Variant-' + trackId;
+                           await (self.rendererType).renderInClient(rpcManager, {
+                              assemblyName,
+                              regions: [region],
+                              adapterConfig: self.adapterConfig,
+                              rendererType: self.rendererType.name,
+                              sessionId: getRpcSessionId(self),
+                              timeout: 1000000,
+                              ...self.renderProps(),
+                           })
+                           self.setReady(true)
+                        } else {
+                           self.setReady(true)
+                        }
+                     } catch (error) {
+                        console.error(error)
+                        self.setError(error)
+                     }
+                  },
+                  { delay: 1000 },
+               ),
+            )
+         },
+         selectFeature(feature){
+            var extendedVariantDisplayConfig
+            if (getContainingTrack(self).configuration.metadata.value.extendedVariantDisplayConfig){
+               extendedVariantDisplayConfig = getContainingTrack(self).configuration.metadata.value.extendedVariantDisplayConfig
```

Comment:
In the JB2 plugin examples, I noticed the getConfg() function, which is documented here:

https://jbrowse.org/jb2/docs/developer_guide/#accessing-config-values

That's supposed to be a preferred way to access config properties, including nested ones. Can you have a look at this and try to convert some of the accessors?

Reply:
Running into some trouble shifting over to getConf / readConfObject. OK to open an issue to convert this after this PR or would you rather it be done here?

Reply:
what kind of issues? i guess if it isnt easily fixable, can you open an issue on this point? As we build the plugin out i want to be sure we dont accumulate too much technical debt, but we can separate from this PR if we need

Reply:
I was getting some issues importing getConf. Not quite sure what was causing it, but I wiped my local changes and tried the same thing again and it's working now, probably was just something small that I missed. Just pushed the getConf changes, if the update works on your end I'd say we're good to go.